en/docs/InstallAndSetup/UpgradingWSO2APIManager/upgrading-from-210-to-300.md

```diff
@@ -1,303 +1,1878 @@
 
-`/api/_UserInfoAPI_.xml`
+        ```
+        [database.apim_db]
+        type = "mssql"
+        url = "jdbc:sqlserver://localhost:1433;databaseName=mig_am_db;SendStringParametersAsUnicode=false"
+        username = "username"
+        password = "password"
+        driver = "com.microsoft.sqlserver.jdbc.SQLServerDriver"
+        validationQuery = "SELECT 1"
+        ```
 
-`/sequences/_auth_failure_handler_.xml`
+    !!! note
+        It is recommended to use the default H2 database for the `WSO2_MB_STORE_DB` database in API-Manager. So do **not** migrate `WSO2_MB_STORE_DB` database from API-M 2.1.0 version to API-M 3.0.0 version, and use the **default H2** `WSO2_MB_STORE_DB` database available in API-M 3.0.0 version.
 
-`/sequences/_build_.xml`
+4.  Update `<API-M_3.0.0_HOME>/repository/resources/conf/default.json` file by pointing to the WSO2UM_DB.
 
-`/sequences/_cors_request_handler_.xml`
+    ```
+    "realm_manager": {
+        "data_source": "WSO2USER_DB",
+        "properties": {
+        "isCascadeDeleteEnabled": true,
+        "initializeNewClaimManager": true
+        }
+    }
+    ```
 
-`/sequences/fault.xml`
+5.  Copy the relevant JDBC driver to the `<API-M_3.0.0_HOME>/repository/components/lib` folder.
 
-`/sequences/main.xml`
+    !!! info
+        In API-M 3.0.0, you do not need to configure the registry configurations as you did in the `<OLD_API-M_HOME>/repository/conf/registry.xml` file and the user database configurations as you did in in the `<OLD_API-M_HOME>/repository/conf/user-mgt.xml` file, as those configurations have been handled internally.
 
-`/sequences/_production_key_error_.xml`
+6.  Move all your Synapse configurations to API-M 3.0.0 pack.
+    -   Move your Synapse super tenant configurations.
+        Copy the contents in the `<OLD_API-M_HOME>/repository/deployment/server/synapse-configs/default` directory and replace the contents in the `<API-M_3.0.0_HOME>/repository/deployment/server/synapse-configs/default` directory with the copied contents.
+    -   Move all your tenant Synapse configurations.
+        Copy the contents in the `<OLD_API-M_HOME>/repository/tenants` directory and replace the contents in the `<API-M_3.0.0_HOME>/repository/tenants` directory with the copied contents.
 
-`/sequences/_resource_mismatch_handler_.xml`
+    !!! warning
+        When moving the Synapse configurations, **do not replace** the following set of files as they contain some modificatiosn in API-M 3.0.0 version.
 
-`/sequences/_sandbox_key_error_.xml`
+        -   /api/_RevokeAPI_.xml
+        -   /sequences/_cors_request_handler_.xml
+        -   /sequences/main.xml
+        -   proxy-services/WorkflowCallbackService.xml
```

Comment:
In 2.1.0 to 2.6.0 migration doc [1], there are list of sequences that should not be replaced in the migrated version. Did we check on that?

[1] https://docs.wso2.com/display/AM260/Upgrading+from+the+Previous+Release#210

Reply:
Yes. There is a discussion happened internally and based on the conclusion, edited the list.